raiden_contracts/deploy/etherscan_verify.py

```diff
@@ -0,0 +1,191 @@
+            source_path['raiden'].joinpath('Utils.sol').read_text() +
+            source_path['raiden'].joinpath('Token.sol').read_text() +
+            source_path['raiden'].joinpath('lib', 'ECVerify.sol').read_text() +
+            source_path['raiden'].joinpath('SecretRegistry.sol').read_text() +
+            source_path['raiden'].joinpath('TokenNetwork.sol').read_text() +
+            source_path['raiden'].joinpath('TokenNetworkRegistry.sol').read_text()
+        )
+
+        source = 'pragma solidity ^0.4.23;' + source.replace('pragma solidity ^0.4.23;', '')
+        source = source.replace('import "raiden/Token.sol";\n', '')
+        source = source.replace('import "raiden/Utils.sol";\n', '')
+        source = source.replace('import "raiden/lib/ECVerify.sol";\n', '')
+        source = source.replace('import "raiden/SecretRegistry.sol";\n', '')
+        source = source.replace('import "raiden/TokenNetwork.sol";\n', '')
+
+        flatten_path = Path(__file__).parent.joinpath('flatten.sol')
+        flatten_path.open('w').write(source)
+
+        constructor_arguments = deployment_info['contracts'][
+            CONTRACT_TOKEN_NETWORK_REGISTRY
+        ]['constructor_arguments']
+        abi = contract_manager.contracts[CONTRACT_TOKEN_NETWORK_REGISTRY]['abi']
+        constructor_types = [
+            arg['type'] for arg in list(
+                filter(lambda x: x['type'] == 'constructor', abi),
+            )[0]['inputs']
+        ]
+        constructor_args = encode_abi(constructor_types, constructor_arguments).hex()
+        print('constructor_args', constructor_arguments, constructor_types, constructor_args)
+
+        etherscan_verify_contract(
+            etherscan_api,
+            apikey,
+            contract_manager.contracts[CONTRACT_TOKEN_NETWORK_REGISTRY],
+            CONTRACT_TOKEN_NETWORK_REGISTRY,
+            deployment_info['contracts'][CONTRACT_TOKEN_NETWORK_REGISTRY],
+            source,
+            constructor_args,
+        )
+
+
+def etherscan_verify_contract(
+        etherscan_api,
+        apikey,
+        compiled_info,
+        contract_name,
+        deployment_info,
+        source_code,
+        constructor_args,
+):
+    metadata = json.loads(compiled_info['metadata'])
+    data = {
+        # A valid API-Key is required
+        'apikey': apikey,
+        # Do not change
+        'module': 'contract',
+        # Do not change
+        'action': 'verifysourcecode',
+        'contractaddress': deployment_info['address'],
+        'sourceCode': source_code,
+        'contractname': contract_name,
+        'compilerversion': 'v' + metadata['compiler']['version'],
+        # 0 = Optimization used, 1 = No Optimization
+        'optimizationUsed': 0 if metadata['settings']['optimizer']['enabled'] is False else 1,
+        'runs': metadata['settings']['optimizer']['runs'],
+        'constructorArguments': constructor_args,
+    }
+    print({k: v for k, v in data.items() if k is not 'sourceCode'})
```

Comment:
Might be a good idea to use `pprint` to have the data printed a bit more readable?